go/chat/sender.go

```diff
@@ -86,28 +87,38 @@ func (s *BlockingSender) addPrevPointersToMessage(ctx context.Context, msg chat1
 
 	var prevs []chat1.MessagePreviousPointer
 
-	res, err := s.G().ConvSource.PullLocalOnly(ctx, convID, msg.ClientHeader.Sender,
+	res, _, err := s.G().ConvSource.Pull(ctx, convID, msg.ClientHeader.Sender,
 		&chat1.GetThreadQuery{
 			DisableResolveSupersedes: true,
 		},
 		&chat1.Pagination{
-			Num: -1,
+			Num: 50,
```

Comment:
I think this is fine. I think the weird corner case that this could affect is:

> There are some unprev'd messages way back in the history. Those might get stuck in unprev'd land forever, once they fall out of the window that we fetch to construct prev ptrs. That could allow the server to move those messages arbitrarily forward in time/msg-id-space (leaving a hole in the message ids), if it's willing to fork the world after that. Pretty unlikely to happen in practice, since you'd have to send 50 messages before a client fetched the first one. Could the server force it to happen?

I don't know the answer to the server force question. The server might be able to play particularly nasty tricks by telling you that a send failed (so that you don't get a msg id for the message in question, and never include it in your own prev pointers), but then actually making the send succeed later, after it's fallen out of the fetch window. It might be that we need an extension to the prev pointer system that allows it to assert send failures. Something to the effect of "I come after messages A, B, and C, and also hey by the way message D was supposedly a send failure, so if you ever see a message with the D body hash then that right there is some _shenanigans_." I haven't thought all that through yet. Feels like the kind of the thing that should be solved with a big hammer, rather than needing a dozen different mitigations for different edge cases...

So yeah, ship it for now :)